MultiLinePlotter/scripts/app.js

```diff
@@ -0,0 +1,106 @@
+var app = angular.module('multiLinePlotterApp', []);
+
+app.controller("app", function ($scope, $http) {
+    $scope.error = null;
+    $scope.isError = false;
+    $scope.data = [];
+    $scope.ykeys = [];
+    $scope.labels = [];
+    $scope.plot = null;
+    $scope.queryRecords = [];
+    $scope.statuses = [];
+    $scope.graphLoading = false;
+    $scope.tweetsLoading = false;
+
+    /* Function to check for error in inputs and call loklak api
+       with the user provided query
+    */
+
+    $scope.search  = function() {
+        $scope.isError = false;
+        if ($scope.tweet === undefined || $scope.tweet === "") {
+            $scope.isError = true;
+            $scope.error = "Please enter a valid query"
+            return;
+        }
+        if ($scope.queryRecords.indexOf($scope.tweet) !== -1) {
+            $scope.isError = true;
+            $scope.error = "Graph has already been plotted for this query"
+            return;
+        }
+        $scope.graphLoading = true;
+        $http.jsonp('http://api.loklak.org/api/search.json?callback=JSON_CALLBACK&q=' +
+            $scope.tweet + '%20since:2015-12-10&source=cache&count=0&fields=created_at')
```

Comment:
and IMHO dealing with request body will be better approach than dealing with strings.
though this same approach I have to implement in my project 😅